gpt4all-bindings/python/docs/gpt4all_python.md

```diff
@@ -91,4 +100,366 @@ To interact with GPT4All responses as the model generates, use the `streaming =
+    who are you?
+    ### Response:
+
+    ===/LLModel.prompt_model -- prompt/===
+    I am an AI-powered chatbot designed to assist users with their queries related to geographical information.
+    INFO:gpt4all.pyllmodel:LLModel.prompt_model -- prompt:
+    ### Instruction:
+    what are your favorite 3 mountains?
+    ### Response:
+
+    ===/LLModel.prompt_model -- prompt/===
+    1) Mount Everest - Located in the Himalayas, it is the highest mountain on Earth and a significant challenge for mountaineers.
+    2) Kangchenjunga - This mountain is located in the Himalayas and is the third-highest peak in the world after Mount Everest and K2.
+    3) Lhotse - Located in the Himalayas, it is the fourth highest mountain on Earth and offers a challenging climb for experienced mountaineers.
+    ```
+
+
+### Without Online Connectivity
+To prevent GPT4All from accessing online resources, instantiate it with `allow_download=False`. This will disable both
+downloading missing models and [models.json], which contains information about them. As a result, predefined templates
+are used instead of model-specific system and prompt templates:
+
+=== "GPT4All Default Templates Example"
+    ``` py
+    from gpt4all import GPT4All
+    model = GPT4All('ggml-mpt-7b-chat.bin', allow_download=False)
+    # when downloads are disabled, it will use the default templates:
+    print("default system template:", repr(model.config['systemPrompt']))
+    print("default prompt template:", repr(model.config['promptTemplate']))
+    print()
+    # even when inside a session:
+    with model.chat_session():
+        assert model.current_chat_session[0]['role'] == 'system'
+        print("session system template:", repr(model.current_chat_session[0]['content']))
+        print("session prompt template:", repr(model._current_prompt_template))
+    ```
+=== "Output"
+    ```
+    default system template: ''
+    default prompt template: '### Human: \n{0}\n### Assistant:\n'
+
+    session system template: ''
+    session prompt template: '### Human: \n{0}\n### Assistant:\n'
+    ```
+
+
+### Interrupting Generation
+The simplest way to stop generation is to set a fixed upper limit with the `max_tokens` parameter.
+
+If you know exactly when a model should stop responding, you can add a custom callback, like so:
+
+=== "GPT4All Custom Stop Callback"
+    ``` py
+    from gpt4all import GPT4All
+    model = GPT4All('orca-mini-3b.ggmlv3.q4_0.bin')
+
+    def stop_on_token_callback(token_id, token_string):
+        # one sentence is enough:
+        if '.' in token_string:
+            return False
+        else:
+            return True
+
+    response = model.generate('Blue Whales are the biggest animal to ever inhabit the Earth.',
+                              temp=0, callback=stop_on_token_callback)
+    print(response)
+    ```
+=== "Output"
+    ```
+     They can grow up to 100 feet (30 meters) long and weigh as much as 20 tons (18 metric tons).
+    ```
+
+
+### FAQ
```

Comment:
We have a FAQ section - maybe we want to put this information here? https://docs.gpt4all.io/gpt4all_faq.html

I'm fine if you don't want to though. Happy to merge with a python specific FAQ here.

Reply:
Right sorry, I was busy with other things until now.

These are kind of specific to the Python bindings and come up frequently enough so that's why I named the section that way. Maybe I shouldn't have called it FAQ?

Or maybe the other FAQ should have a subsection for bindings? I'm kind of unsure about it, it makes sense in either.